### PR TITLE
test: add `insta` snapshots to borsh/tests

### LINKED ISSUE #152: Introduce snapshot testing for all the serialized results
Currently, borsh tests only ensure that serialization is reversible if used with exactly the same version of borsh on the same system:

https://github.com/near/borsh-rs/blob/4f8f985a316c0d6c06d0e393a207ee8d6537f01a/borsh/tests/test_primitives.rs#L8-L10

This won't catch if the format suddenly changes on both the serialization and deserialization sides.

I have really great experience with [`insta` crate](https://github.com/mitsuhiko/insta), which helps to set up snapshot testing in no time.

Basically, I suggest we add an extra line in almost every test:

```rust
insta::assert_debug_snapshot!(buf);
```

This will allow us to instantly spot if the underlying format gets changed.

### PR DESCRIPTION
**Pull request by dj8yfo**
Resolves #152

### CODE REVIEW

**Comment by dj8yfo** [comment]
- [x] borsh::test_arrays:
- [x] borsh::test_tuple:
- [x] borsh::test_primitives:
- [x] borsh::test_bson_object_ids:
- [x] borsh::test_vecs:
- [x] borsh::test_strings:
- [x] borsh::test_hash_map:
- [ ] borsh::test_schema_enums:
  - [ ] complex_enum_with_schema?
- [x] borsh::test_simple_structs:
- [x] borsh::test_generic_struct:
- [x] borsh::test_binary_heaps:

File: borsh/tests/test_schema_enums.rs [hunk 1]
```diff
@@ -148,6 +148,7 @@ pub fn complex_enum_with_schema() {
     // Then check that we serialize and deserialize with schema.
     let obj = A::default();
     let data = try_to_vec_with_schema(&obj).unwrap();
+    // insta::assert_debug_snapshot!(buf);
```
Comment (frol) [inline borsh/tests/test_schema_enums.rs:151]: Why is it commented out? Is it too big? Is it not stable?

**Review by frol — CHANGES_REQUESTED** [review]
@dj8yfo Looks good to me! Let's rebase/resolve merge conflict and address the only comment I left before merging.

File: borsh/tests/test_schema_enums.rs (hunk 1 above)
Comment (dj8yfo) [reply borsh/tests/test_schema_enums.rs]: I've mentioned that this example of schema derivation https://github.com/near/borsh-rs/blob/master/borsh-schema-derive-internal/src/enum_schema.rs#L365 is suboptimal, as structs representing enum variants depend on more generic params than their counterparts in original enum. 
There may be more of such details, which will affect schema serialization.

File: borsh/tests/test_simple_structs.rs [hunk 2]
```diff
@@ -171,6 +171,8 @@ fn test_simple_struct() {
     let e = E { a: &a };
     let encoded_ref_a = e.try_to_vec().unwrap();
     assert_eq!(encoded_ref_a, encoded_a);
+    #[cfg(feature = "std")]
+    insta::assert_debug_snapshot!(encoded_a);
```
Comment (frol) [inline borsh/tests/test_simple_structs.rs:175]: This is a pity that we skip the format safety checks for no-std. Is there anything we can do about it? 🤔

Comment (dj8yfo) [reply borsh/tests/test_simple_structs.rs:175]: nope, not with `insta` crate . it explicitly imports `std` in expanded macros

Comment (dj8yfo) [reply borsh/tests/test_simple_structs.rs:175]: https://crates.io/crates/expect-test and https://crates.io/crates/k9 and don't feature `no_std` as far as i can see as well

Comment (frol) [reply borsh/tests/test_simple_structs.rs:175]: Ok, let's live with this for now. Maybe someone will contribute no_std support to insta or some other similar tool.

**Review by frol — APPROVED** [review]

**Pull request merged**